src/NuGet.Core/NuGet.Protocol/Plugins/PluginCacheEntry.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using NuGet.Common;
+
+namespace NuGet.Protocol.Plugins
+{
+    public sealed class PluginCacheEntry
```

Comment:
nit: Will `PluginCacheUtility` be a better choice than having `PluginCacheEntry` and instantiating it for every requestkey? You essentially need a utility class with couple of static apis like `HttpCacheUtility`

Reply:
Hmm..I thought about that, but then I end up passing the cache file name, root folder more often in those APIs, and it's more error prone that way. 

Even in the http caching utility...there's still a cached result, and a caching context etc. 
Here, it's all combined in one model.